interchain-token-service/src/user_functions.rs

```diff
@@ -385,10 +294,19 @@ pub trait UserFunctionsModule:
                 );
                 require!(second_payment.amount == gas_amount, "Invalid gas value");
```

Comment:
`second_payment` could be none here.

Reply:
It can not, there is an `if second_payment.is_none()` above that does an early return.